src/librustdoc/core.rs

```diff
@@ -570,6 +587,75 @@ pub fn run_core(options: RustdocOptions) -> (clean::Crate, RenderInfo, RenderOpt
     })
 }
 
+use rustc_hir::def::Res;
+use rustc_hir::{
+    intravisit::{NestedVisitorMap, Visitor},
+    Path,
+};
+use rustc_middle::hir::map::Map;
+
+thread_local!(static DEFAULT_TYPECK: for<'tcx> fn(rustc_middle::ty::TyCtxt<'tcx>, rustc_span::def_id::LocalDefId) -> &'tcx rustc_middle::ty::TypeckTables<'tcx> = {
+    let mut providers = rustc_middle::ty::query::Providers::default();
+    rustc_typeck::provide(&mut providers);
+    providers.typeck_tables_of
+});
+
+/// Due to https://github.com/rust-lang/rust/pull/73566,
+/// the name resolution pass may find errors that are never emitted.
+/// If typeck is called after this happens, then we'll get an ICE:
+/// 'Res::Error found but not reported'. To avoid this, emit the errors now.
+struct EmitIgnoredResolutionErrors<'a, 'hir> {
+    session: &'a Session,
+    hir_map: Map<'hir>,
+}
+
+impl<'a, 'hir> EmitIgnoredResolutionErrors<'a, 'hir> {
+    fn new(session: &'a Session, hir_map: Map<'hir>) -> Self {
+        Self { session, hir_map }
+    }
+}
+
+impl<'hir> Visitor<'hir> for EmitIgnoredResolutionErrors<'_, 'hir> {
+    type Map = Map<'hir>;
+
+    fn nested_visit_map(&mut self) -> NestedVisitorMap<Self::Map> {
+        // We need to recurse into nested closures,
+        // since those will fallback to the parent for type checking.
+        NestedVisitorMap::OnlyBodies(self.hir_map)
+    }
+
+    fn visit_path(&mut self, path: &'v Path<'v>, _id: HirId) {
+        log::debug!("visiting path {:?}", path);
+        if path.res == Res::Err {
+            // We have less context here than in rustc_resolve,
+            // so we can only emit the name and span.
+            // However we can give a hint that rustc_resolve will have more info.
+            // NOTE: this is a very rare case (only 4 out of several hundred thousand crates in a crater run)
+            // NOTE: so it's ok for it to be slow
+            let label = format!(
+                "could not resolve path `{}`",
+                path.segments
+                    .iter()
+                    .map(|segment| segment.ident.as_str().to_string())
+                    .collect::<Vec<_>>()
+                    .join("::")
+            );
+            let mut err = rustc_errors::struct_span_err!(
+                self.session,
+                path.span,
+                E0433,
+                "failed to resolve: {}",
+                label
+            );
+            err.span_label(path.span, label);
+            err.note("this error was originally ignored because you are running `rustdoc`");
+            err.note("try running again with `rustc` and you may get a more detailed error");
+            err.emit();
+        }
+        // NOTE: this does _not_ visit the path segments
```

Comment:
Doesn't this mean that it won't catch `Vec<DoesNotExist>`?

Reply:
Ahhh, good catch. That gave the delay_span_bug ICE. I'll fix it and add a test case.

```
error: internal compiler error: TyKind::Error constructed but no error reported
  |
  = note: delayed at /home/joshua/src/rust/src/librustc_session/session.rs:436:9
```